ledger/src/shred.rs

```diff
@@ -467,8 +473,36 @@ impl Shred {
         self.common_header.signature
     }
 
-    pub fn seed(&self) -> [u8; 32] {
+    pub fn seed(
+        &self,
+        leader_schedule_cache: &Arc<LeaderScheduleCache>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+    ) -> [u8; 32] {
         let mut seed = [0; 32];
+
+        let bf = bank_forks.read().unwrap();
+        let maybe_bank = bf.get(self.slot());
+        if let Some(bank) = maybe_bank {
+            if enable_deterministic_seed(self.slot(), &bank) {
+                let maybe_leader_pubkey =
+                    leader_schedule_cache.slot_leader_at(self.slot(), Some(&bank));
+                let h = match maybe_leader_pubkey {
```

Comment:
Now would probably be a good time to add hash domain prefix strings to these values. Something like `"solana shred seed v1"` as the first input to the hash would be great.  Are the leader-available and not cases expected to co-exist? Their prefixes should be different if so.

cc/ @jbiseda

Reply:
What would be the use of the hash-prefix here?

Reply:
Any unique string would be fine.  I suggested 
* `b"solana shred seed v1"`

above.  If we need to separate the cases, 
* `b"solana shred seed v1 with leader"`
* `b"solana shred seed v1 without leader"`

would be my suggestion